src/main/java/Warehouse/WarehouseManager/employee/EmployeeService.java

```diff
@@ -0,0 +1,50 @@
+package Warehouse.WarehouseManager.employee;
+
+import Warehouse.WarehouseManager.exception.EmailAlreadyExistsException;
+import Warehouse.WarehouseManager.exception.EmployeeNotExistsException;
+import Warehouse.WarehouseManager.exception.EmptyDataException;
+import Warehouse.WarehouseManager.exception.UsernameAlreadyExistsException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+public class EmployeeService {
+
+    private EmployeeRepository employeeRepository;
+    private final BCryptPasswordEncoder bCryptPasswordEncoder;
+
+    @Autowired
+    public EmployeeService(EmployeeRepository employeeRepository, BCryptPasswordEncoder bCryptPasswordEncoder) {
+        this.employeeRepository = employeeRepository;
+        this.bCryptPasswordEncoder = bCryptPasswordEncoder;
+    }
+    public EmployeeDto getEmployeeByUsername(String username){
+        Employee employee = employeeRepository.findEmployeeByUsername(username)
+                .orElseThrow(() -> new EmployeeNotExistsException(username));
+        return employee.toEmployeeDto();
+    }
+
+    @Transactional
+    public EmployeeDto employeeRegistration(EmployeeDto employeeDto){
```

Comment:
Do we need to return EmployeeDto here?

Reply:
The method returns EmployeeDto simplify testing, since the method returns a value, we have a better chance of catching data conversion errors.